src/data/energy.ts

```diff
@@ -1376,22 +1376,54 @@ export const getPowerFromState = (stateObj: HassEntity): number | undefined => {
     return undefined;
   }
 
-  // Normalize to kW based on unit of measurement (case-sensitive)
+  // Normalize to watts (W) based on unit of measurement (case-sensitive)
   // Supported units: GW, kW, MW, mW, TW, W
   const unit = stateObj.attributes.unit_of_measurement;
   switch (unit) {
     case "W":
-      return value / 1000;
+      return value;
+    case "kW":
+      return value * 1000;
     case "mW":
-      return value / 1000000;
+      return value / 1000;
     case "MW":
-      return value * 1000;
+      return value * 1_000_000;
     case "GW":
-      return value * 1000000;
+      return value * 1_000_000_000;
     case "TW":
-      return value * 1000000000;
+      return value * 1_000_000_000_000;
     default:
-      // Assume kW if no unit or unit is kW
+      // Assume value is in watts (W) if no unit or an unsupported unit is provided
       return value;
   }
 };
+
+/**
+ * Format power value in watts (W) to a short string with the appropriate unit
+ * @param hass - The HomeAssistant instance
+ * @param powerWatts - The power value in watts (W)
+ * @returns A string with the formatted power value and unit
+ */
+export const formatPowerShort = (
+  hass: HomeAssistant,
+  powerWatts: number
+): string => {
+  const units = ["W", "kW", "MW", "GW", "TW"];
+  let unitIndex = 0;
+  let value = powerWatts;
+
+  // Scale the unit to the appropriate power of 1000
+  while (Math.abs(value) >= 1000 && unitIndex < units.length - 1) {
+    value /= 1000;
+    unitIndex++;
+  }
+
+  return (
+    formatNumber(value, hass.locale, {
+      // For watts, show no decimals. For kW and above, always show 3 decimals.
+      maximumFractionDigits: units[unitIndex] === "W" ? 0 : 3,
+    }) +
+    " " +
```

Comment:
We generally use `blankBeforeUnit` for this but it only has an effect on percent and degrees so I don't think it's worth it here